src/Multicall.sol

```diff
@@ -21,7 +21,7 @@ contract Multicall {
                 let _retLen     := mul(mload(add(data, add(cur, 0x20))), 0x20)
                 let _dataLength := mload(add(data, add(cur, 0x60)))
                 let _data       := add(data, add(cur, 0x80))
-                if eq(call(gas, _target, 0, _data, _dataLength, ptr, _retLen), 0)
+                if eq(staticcall(gas, _target, _data, _dataLength, ptr, _retLen), 0)
```

Comment:
@gbalabasquer The issue with using `staticcall` instead of `call` is that we won't be able to call functions that _may_ change state before returning a value (like [chi](https://github.com/makerdao/sai/blob/0dd0a799e4746ac1955b67898762cff9b71aea17/src/tub.sol#L203) for example).

Reply:
Yes you are right. It seems the only solution is like you did in the other PR removing the `view` modifier. This is a bit dangerous, but I guess we do not have many options. Considering `multicall` will be mostly called from the js package, it can be handled.
However probably is better to keep the abi as it is for now, so we can avoid 100% of warnings, and we also don't need to do any major changes in the multicall.js (just making sure to execute the call as a static one).
I've pushed a new commit changing back the `staticcall` to `call` and I've also managed to remove the `BytesLib` at all.  This commit is 100% warnings free.

Reply:
Nice, much cleaner tests. :)

I'm going to merge this in, but it might be worth us moving to the `ABIEncoderV2` approach soon. The warnings are annoying, but we get code that's more easy to read and audit, and variable length return data (which is a really nice feature).

And because we'll only be making read-only calls, I think the warnings about using `ABIEncoderV2` in production are a non-issue.

Reply:
Yeah you are right. Actually `dss` is using it. It will not bring any bug, just trying to keep most of the contracts with clean outputs in the scripts. As `multicall` can be upgraded any time is why I think we could just wait for the `ABIEncoderV2 ` not being experimental. But maybe you are right and we should just upgrade it.